Sources/LyftUI/LyftDeepLink.swift

```diff
@@ -2,57 +2,90 @@ import Foundation
+    /// - parameter behavior:       The deep linking mode to use for this deep link
+    /// - parameter kind:           The kind of ride to create a request for
+    /// - parameter pickup:         The pickup position of the ride
+    /// - parameter destination:    The destination position of the ride
+    /// - parameter couponCode:     A coupon code to be applied to the user
+    public static func requestRide(using behavior: LyftDeepLinkBehavior = .native, kind: RideKind = .Standard,
                                    from pickup: CLLocationCoordinate2D? = nil,
                                    to destination: CLLocationCoordinate2D? = nil,
                                    couponCode: String? = nil)
     {
-        var parameters = ["id": kind.rawValue]
+        let action: String
+        var parameters = [String: Any]()
+        parameters["partner"] = LyftConfiguration.developer?.clientId
+        parameters["credits"] = couponCode
 
-        if let pickup = pickup {
-            parameters["pickup[latitude]"] = String(pickup.latitude)
-            parameters["pickup[longitude]"] = String(pickup.longitude)
-        }
+        switch behavior {
+            case .native:
+                action = "ridetype"
+                parameters["id"] = kind.rawValue
+                parameters["pickup[latitude]"] = pickup.map { $0.latitude }
+                parameters["pickup[longitude]"] = pickup.map { $0.longitude }
+                parameters["destination[latitude]"] = destination.map { $0.latitude }
+                parameters["destination[longitude]"] = destination.map { $0.longitude }
 
-        if let destination = destination {
-            parameters["destination[latitude]"] = String(destination.latitude)
-            parameters["destination[longitude]"] = String(destination.longitude)
+            case .web:
+                action = "request"
+                parameters["ride_type"] = kind.rawValue
+                parameters["pickup"] = pickup.map { "@\($0.latitude),\($0.longitude)" }
+                parameters["destination"] = destination.map { "@\($0.latitude),\($0.longitude)" }
         }
 
-        parameters["partner"] = LyftConfiguration.developer?.clientId
-        parameters["credits"] = couponCode
-
-        self.launch(action: "ridetype", parameters: parameters)
+        self.launch(using: behavior, action: action, parameters: parameters)
     }
 
-    private static func launch(action: String, parameters: [String: String]?) {
-        var requestURLComponents = URLComponents()
-        requestURLComponents.scheme = "lyft"
-        requestURLComponents.host = action
-        requestURLComponents.queryItems = parameters?.map { key, value in
-            URLQueryItem(name: key, value: value)
+    private static func launch(using behavior: LyftDeepLinkBehavior, action: String,
+                               parameters: [String: Any])
+    {
+        guard let baseUrl = URL(string: behavior.baseUrl + action) else {
```

Comment:
Why is `behavior.baseUrl` not an url? I'd make it an URL and use `URL(string:relativeTo:)` here